@ngrx/router-store update to 9.1.1 causes compilation error.

The type of selectRouteParam appear to have changed between @ngrx/router-store@9.1.0 and @ngrx/router-store@9.1.1.  Using 9.1.0 on my project everything is fine, after updating to 9.1.1 I am getting a compilation error.

## Minimal reproduction of the bug/regression with instructions:

```Typescript
export const selectRouter = createFeatureSelector<RouterReducerState<MinimalRouterStateSnapshot>>(
  DEFAULT_ROUTER_FEATURENAME
);

export const {
  selectQueryParams,
  selectQueryParam,
  selectRouteParams,
  selectRouteParam,
  selectRouteData,
  selectUrl
} = getSelectors(selectRouter);

export const selectPath = selectRouteParam('path');

export interface Entity {
  id: string;
  path: string;
}

@Injectable({ providedIn: 'root' })
export class EntityService extends EntityCollectionServiceBase<Entity> {
  constructor(serviceElementsFactory: EntityCollectionServiceElementsFactory) {
    super('Entity', serviceElementsFactory);
  }
  private readonly selectCurrentEntity = createSelector(
    this.selectors.selectEntities,
    selectPath,
    (entities, path) => entities.find(x => x.path === path) || null
  );
}
```
When I use 9.1.1 I get this error
```
ERROR in src/app/data/entity/entity.service.ts:21:38 - error TS2367: This condition will always return 'false' since the types 'string' and 'MemoizedSelector<object, string | undefined, DefaultProjectorFn<string | undefined>>' have no overlap.

21     (entities, path) => entities.find(x => x.path === path) || null
                                              ~~~~~~~~~~~~~~~
```
When I use 9.1.0 the compilation is successful and the type of path is ```string | undefined```.

I have created [a project containing this code on stackblitz](https://stackblitz.com/edit/ngrx-seed-u45tgp) to demonstrate this issue.
## Expected behavior:
It should compile cleanly under 9.1.1 as it does when using 9.1.0.
## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):
```
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 9.1.4
Node: 12.16.3
OS: linux x64

Angular: 9.1.4
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router, service-worker
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.901.4
@angular-devkit/build-angular      0.901.4
@angular-devkit/build-optimizer    0.901.4
@angular-devkit/build-webpack      0.901.4
@angular-devkit/core               9.1.4
@angular-devkit/schematics         9.1.4
@angular/cdk                       9.2.2
@angular/flex-layout               9.0.0-beta.29
@angular/material                  9.2.2
@angular/material-moment-adapter   9.2.2
@ngtools/webpack                   9.1.4
@schematics/angular                9.1.4
@schematics/update                 0.901.4
rxjs                               6.5.5
typescript                         3.8.3
webpack                            4.42.0
```
## Other information:

Only @ngrx/router-store needs to be reverted to 9.1.0 to fix the issue, but I have not tested that my application works correctly when the other components are 9.1.1, but @ngrx/router-store is at 9.1.0.

## I would be willing to submit a PR to fix this issue

[X] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

--- a/modules/router-store/src/models.ts
+++ b/modules/router-store/src/models.ts
@@ -1,14 +1,11 @@
 import { Data, Params } from '@angular/router';
-import { MemoizedSelector } from '@ngrx/store';
 
 export interface RouterStateSelectors<V> {
-  selectCurrentRoute: MemoizedSelector<V, any>;
-  selectQueryParams: MemoizedSelector<V, Params>;
-  selectQueryParam: (param: string) => MemoizedSelector<V, string | undefined>;
-  selectRouteParams: (state: V) => MemoizedSelector<V, Params>;
-  selectRouteParam: (
-    param: string
-  ) => (state: V) => MemoizedSelector<V, string | undefined>;
-  selectRouteData: (state: V) => MemoizedSelector<V, Data>;
-  selectUrl: (state: V) => MemoizedSelector<V, string>;
+  selectCurrentRoute: (state: V) => any;
+  selectQueryParams: (state: V) => Params;
+  selectQueryParam: (param: string) => (state: V) => string | undefined;
+  selectRouteParams: (state: V) => Params;
+  selectRouteParam: (param: string) => (state: V) => string | undefined;
+  selectRouteData: (state: V) => Data;
+  selectUrl: (state: V) => string;
 }

